include/oneapi/dpl/internal/serial_numeric_impl.h

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) Intel Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef _ONEDPL_SERIAL_NUMERIC_IMPL_H
+#define _ONEDPL_SERIAL_NUMERIC_IMPL_H
+#include <iterator>
+namespace oneapi
+{
+namespace dpl
+{
+// libstdc++7 and libstdc++8 doesn't include reduce;
+#if (__cplusplus < 201703L) || (_GLIBCXX_RELEASE < 9)
```

Comment:
Our own implementation for reduce can only work when system std doesn't include implementation for reduce. This happens when:
1. developers are using c++14, c++11 or more legacy c++ verions.
2. If developers are working in linux platform, they are using libstdc++7,8 since libstdc++7/8 does not include impl for reduce.
If the system std library in use already includes impl for reduce, we can't provide our own impl otherwise conflicts will happen.
Hi, @capatober 
Could you check if the logic here is correct?
Thanks very much.